src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    public struct HashCode
+    {
+#   if SYSTEM_HASHCODE_TESTVECTORS
+        private static readonly uint _seed = 0;
+#   else
+        private static readonly uint _seed = unchecked((uint)new Random().Next(int.MinValue, int.MaxValue));
+#   endif
+
+        private const uint Prime1 = 2654435761U;
+        private const uint Prime2 = 2246822519U;
+        private const uint Prime3 = 3266489917U;
+        private const uint Prime4 = 0668265263U;
+        private const uint Prime5 = 0374761393U;
+
+        private uint _v1, _v2, _v3, _v4;
+        private uint _queue1, _queue2, _queue3;
+        private uint _length;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int Combine<T1>(T1 value1)
+        {
+            unchecked
```

Comment:
Nit: We do not use `unchecked` scopes in corefx.

Reply:
Sure we do, effectively explicitly indicating that overflow or underflow are being accounted for in the code within. (Or when required for a const cast problem)

Reply:
Majority of the code does not use them. We had long discussion about this in https://github.com/dotnet/corefx/pull/16416. But if you think that it adds value here, I do not have problem with it.

Reply:
Looking at the code now (vs noticing the comment in email) I agree that most of these don't add value.  The `Initialize` (somewhat) and Round (yeah) and QueueRound  (okay) and MixState (which currently isn't decorated) it pre-empts the "Is overflow a concern here for interesting values of `s_seed`?" type questions.  Following through that other issue's thread, it's Steve's "category 1" and Karel's "2c": hash functions indicating that overflow is built-in to their calculations.

While `+= 4` could overflow, it doesn't trip my "is this interesting?" nearly as much as adding a random number to a large number (e.g. Initialize) or a lot of `*=` (like Round)

Reply:
Force of habit - I've worked on a few default checked projects which always make me think about overflows and documenting the intent.

Removed.